From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Seungchul Kim <sc377.kim@samsung.com>
Date: Tue, 24 Mar 2020 16:03:54 +0900
Subject: FROMLIST: media: v4l2-fh: define v4l2_fh struct regardless of
 condition

v4l2_fh struct define differently by CONFIG_V4L2_MEM2MEM_DEV.
If some vendors use CONFIG_V4L2_MEM2MEM_DEV by module,
it can make the mismatch of v4l2_fh sturct.

By the mismatch, the following error occurs.
===============================
[    7.533506] v4l2_mem2mem: disagrees about version of symbol video_devdata
[    7.533594] v4l2_mem2mem: Unknown symbol video_devdata (err -22)
[    7.535319] v4l2_mem2mem: disagrees about version of symbol v4l2_event_pending
[    7.542532] v4l2_mem2mem: Unknown symbol v4l2_event_pending (err -22)
===============================

So v4l2_fh struct is modified to does not have dependency
for CONFIG_V4L2_MEM2MEM_DEV.

Bug: 151399858
Link: https://lore.kernel.org/linux-media/000601d600bb$0e2d4320$2a87c960$@samsung.com/T/#u

Change-Id: I6481516d5197988e800853a491474b374ed61d36
Signed-off-by: Seungchul Kim <sc377.kim@samsung.com>
---
 include/media/v4l2-fh.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/media/v4l2-fh.h b/include/media/v4l2-fh.h
index 53b4dbb4ae8e..b5b3e00c8e6a 100644
--- a/include/media/v4l2-fh.h
+++ b/include/media/v4l2-fh.h
@@ -53,9 +53,7 @@ struct v4l2_fh {
 	unsigned int		navailable;
 	u32			sequence;
 
-#if IS_ENABLED(CONFIG_V4L2_MEM2MEM_DEV)
 	struct v4l2_m2m_ctx	*m2m_ctx;
-#endif
 };
 
 /**
